iroh-store/src/store.rs

```diff
@@ -367,6 +400,75 @@ impl<'a> LocalStore<'a> {
         Ok(())
     }
 
+    /// Takes a list of cids and gives them ids, which are boths stored and then returned.
+    #[tracing::instrument(skip(self, cids))]
+    fn ensure_id_many<I>(&mut self, cids: I) -> Result<Vec<u64>>
+    where
+        I: IntoIterator<Item = Cid>,
+    {
+        let mut ids = Vec::new();
+        let mut batch = WriteBatch::default();
+        for cid in cids {
+            let id_key = id_key(&cid);
+            let id = if let Some(id) = self.db.get_pinned_cf(self.cf.id, &id_key)? {
+                u64::from_be_bytes(id.as_ref().try_into()?)
+            } else {
+                let id = self.next_id();
+                let id_bytes = id.to_be_bytes();
+
+                let metadata = MetadataV0 {
+                    codec: cid.codec(),
+                    multihash: cid.hash().to_bytes(),
+                };
+                let metadata_bytes = rkyv::to_bytes::<_, 1024>(&metadata)?; // TODO: is this the right amount of scratch space?
+                batch.put_cf(&self.cf.id, id_key, id_bytes);
+                batch.put_cf(&self.cf.metadata, id_bytes, metadata_bytes);
+                id
+            };
+            ids.push(id);
+        }
+        self.db.write(batch)?;
+
+        Ok(ids)
+    }
+
+    #[tracing::instrument(skip(self))]
+    fn next_id(&mut self) -> u64 {
+        let id = *self.next_id;
+        *self.next_id = id + 1;
+        // TODO: better handling
+        assert!(id > 0, "this store is full");
```

Comment:
Is the overflow behaviour defined or UB?  IIRC debug builds will panic on overflow or do I misremember?  Personally I'd have used `.checked_add()` with an explicit branch to `panic!()` as then I don't have to answer those questions :)

Reply:
Yes, debug will panic on overflow, release will wraparound. I can change it to a checked add.

Reply:
Fixed it as you suggested.